targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F437xG/TARGET_UBLOX_C030/battery_charger_i2c.cpp

```diff
@@ -0,0 +1,33 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2013 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "battery_charger_i2c.h"
+
+BatteryChargerI2c::BatteryChargerI2c(PinName sda, PinName scl):_i2c(), _hz(100000) {
+    i2c_init(&_i2c, sda, scl);
+}
+
+bool BatteryChargerI2c::read_from_i2c(int i2c_address, char* data_read, int length)
+{
+    int bytes_read = i2c_read(&_i2c, i2c_address, data_read, length, 1);
+    return ((length == bytes_read) ? (true) : (false));
```

Comment:
This could be simplified to:

`return (length == bytes_read);`

The compiler will do the rest!

Reply:
Good point